osmtogtfs/osm/builders/stop_builder.py

```diff
@@ -0,0 +1,32 @@
+"""Functionality to build a list of stops."""
+from osmtogtfs.osm.models import Stop
+
+
+def build_stops(relations, nodes):
+    visited_stops_ids = set()
+    for rel in relations.values():
+        for stop in extract_stops(rel, nodes, visited_stops_ids):
+            if stop:
+                yield stop
+
+
+def extract_stops(relation, nodes, visited_stops_ids):
+    """Extract stops in a relation."""
+    for member_id, member_role in relation.member_info:
+        if member_id not in visited_stops_ids and \
+            _is_stop(member_id, member_role, nodes):
+            visited_stops_ids.add(member_id)
+            if member_id in nodes:
+                yield Stop(member_id,
+                    nodes[member_id].tags.get('name') or\
+                    "Unnamed {} stop.".format(relation.tags.get('route')),
+                    nodes[member_id].lon if member_id in nodes else '',
```

Comment:
@mehdisadeghi isnt this redundant? the upper if guarantees that you only get here if the member id is in nodes. So there shouldnt be a reason to double check it here. Or am I reading this wrong?

Reply:
Yes apparently this is duplicate. However, I have to make this explicit. I mean there are always stops that we don't have their corresponding nodes.